test/ts/unit/collateralized/runners/register_term_start.ts

```diff
@@ -0,0 +1,160 @@
+// modules
+import { expect } from "chai";
+import * as ABIDecoder from "abi-decoder";
+import { compact } from "lodash";
+
+// wrappers
+import { MockCollateralizedTermsContractContract } from "types/generated/mock_collateralized_terms_contract";
+import { MockDebtRegistryContract } from "types/generated/mock_debt_registry";
+import { MockERC20TokenContract } from "types/generated/mock_e_r_c20_token";
+import { MockTokenRegistryContract } from "types/generated/mock_token_registry";
+
+// scenarios
+import { RegisterTermStartScenario, TestAccounts, TestContracts } from "./";
+
+// utils
+import { REVERT_ERROR } from "../../../test_utils/constants";
+
+// logs
+import { CollateralLocked } from "../../../logs/collateralized_contract";
+
+export class RegisterTermStartRunner {
+    private contracts: TestContracts;
+    private accounts: TestAccounts;
+
+    constructor() {
+        this.testScenario = this.testScenario.bind(this);
+    }
+
+    public initialize(contracts: TestContracts, accounts: TestAccounts) {
```

Comment:
What's the reasoning for a separate `initialize` function in addition to the constructor.

Reply:
We want to be able to dynamically define tests at runtime using the `testScenario` function (see [here](https://github.com/dharmaprotocol/charta/pull/45/files/f45b0c07125b7ecd0570eef7720f47b85cf44a46#diff-dcf3fbad7b6f2a54aa5b32a8c722cd17R242)).

The above line is evaluated at runtime and will only evaluate correctly in typescript if the `RegisterTermStartRunner` is defined at runtime.  However, our contracts and accounts are pulled asynchronously -- i.e., they're only available after the first `before` in [the tests](https://github.com/dharmaprotocol/charta/pull/45/files/f45b0c07125b7ecd0570eef7720f47b85cf44a46#diff-dcf3fbad7b6f2a54aa5b32a8c722cd17R74).  Therefore, we leverage this admittedly wonky pattern because 

1. We need the RegisterTermStartRunner object to be constructed by the time the tests are dynamically defined.
2. We need the asynchronously pulled contracts to be accessible to the RegisterTermStartRunner once the dynamically defined tests are eventually run.

Keeping as is unless you have alternative suggestions for how to make this cleaner.